keps/sig-cloud-provider/2025-extend-konnectivity-for-both-directions/README.md

```diff
@@ -0,0 +1,694 @@
+
+### Listening Interface for Konnectivity agent
+
+As mentioned before, we will be using the Kubernetes default service to route traffic to the agent. The service in itself has a couple of limitations: it can’t be used as a type externalName, thus preventing usage of DNS names. But also, some general services limitations apply: endpoints can't use the link-local range and the localhost range. This means that we are left with the 3 private IPs ranges (10.0.0.0/8, 172.16.0.0/12 and 192.168.0.0/16).
+
+The agent will create a dummy interface, assign it the ip provided with the `bind-address` flag, using `host` scope, and will start listening on this IP:local_port (local_port is defined with the `target` flag). This will allow all agents to bind to the ip address advertised by the KAS, that will be valid only inside the node.
+
+### Authentication between Konnectivity agent and server
+
+Konnectivity agent currently support mTLS or Token based authentication. Note that API objects such as Secrets cannot be accessed neither when static pod nor systemd service deployment strategy is used. The authentication secret will be made available to the agent through a different channel (e.g. provisioned in the worker node file-system at creation time).
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+#### Allowed Destination
+
+> A generic TCP tunnel is fraught with security risks. First, such
+> authorization should be limited to a small number of known ports.
+> The Upgrade: mechanism defined here only requires onward tunneling at
+> port 80. Second, since tunneled data is opaque to the proxy, there
+> are additional risks to tunneling to other well-known or reserved
+> ports. A putative HTTP client CONNECTing to port 25 could relay spam
+> via SMTP, for example.
+> -- <cite>[rfc2817][http_upgrade_tls_rfc]</cite>
+
+Similarly to CONNECT tunnel this feature opens the door to generic TCP tunnels to arbitrary destinations.
+
+A mechanism for restricting access to the possible destinations on the Master Network should be implemented on the Konnectivity server.
+The allowed destination will be provided with the following flag:
+
+* `--allowed-destination=dst_host:dst_port`: The address and port of the KAS.
+
+Note: if this feature will be extended to allow reaching arbitrary destinations in the master network, this can be easily generalized by allowing multiple occurrences of this flag and maintaining a list of allowed destinations.
```

Comment:
Can we explicitly specify that if this flag is absent the server will no allow any node network initiated requests (traffic) to be placed on the master network.